### DISCUSSION TITLE
Lume Relations does not work?

### DISCUSSION THREAD

**Discussion by officialrajdeepsingh** [lumeland/lume#309] [Q&A] [answered]
When I use the following code on my website, it doesn't show anything.

```javascript
site.use(
relations({
  foreignKeys: {
    article: "article_id",

    author: "author", // replace to the author_id

  }
}));
```
In my posts, I already mention the author meta tag.
```markdown
---
# posts/the-millionaire-guide-on-programing-to-help-you-get-rich.md

title: The Millionaire Guide On Programming To Help You Get Rich.
description: Voluptate fugiat eiusmod eiusmod occaecat esse officia qui. Elit elit velit
  cillum eiusmod aliquip ullamco exercitation sin.
date: 2022-11-08T08:26:45.204Z
draft: false
tags:
  - Millionaire
  - Programming
  - Rich
category:
  - Programming
image:  /images/computer.jpg
author: Jeffery Schwartz
slug: millionaire-guide-programing-rich
---
Mollit enim excepteur rest of article .....
```
In my authors, I already mention the author meta tag.
```markdown
# authors/annie-dean.md
---
author: Annie Dean
description: Sit culpa elit excepteur ea Lorem ullamco eiusmod ex ut ea incididunt minim. Cillum eiusmod fugiat cupidatat.
date: 2022-11-08T09:01:53.077Z
draft: false
image: "/images/female1.jpg"
social: 
  - facebook.com
  - Google.com
job : "Writer"
---

Sit culpa elit excepteur ea rest articles....
```
With `article_id`, it works fine, but it is a problem to remember the id every time. So I replace it with the author's name, but it is not working. 

```javascript
site.use(
relations({
  foreignKeys: {
    article: "article_id",

    author: "article_id",

  }
}));
```
The relations foreign Keys only accept the integer value as an id?

**Comment by oscarotero** [comment, marked as answer]
In your first example:
```js
site.use(
relations({
  foreignKeys: {
    article: "article_id",
    author: "author",
  }
}));
```

The problem is `author` key and value cannot be equal, because the plugin creates a new `author` property that overrides your `author` value. The id can be anything, it's not limited to integers. Probably something like this should work:

```js
site.use(
relations({
  foreignKeys: {
    article: "article_id",
    author: "authorName",
  }
}));
```

If you don't want to use the `id` field to link the pages, you can change it to any other value, but note that this will affect to all relations (cannot be configured individually per relation). Example:

```js
site.use(
relations({
  idKey: "slug",
  foreignKeys: {
    article: "article_slug",
    author: "author_slug",
  }
}));
```

In this example, I've configured the `idKey` as "slug" so all related pages must have a `slug` key. And, for consistency, I've changed the foreign keys names to include `_slug` as suffix.

**Reply by oscarotero** [reply]
Yes, sure! You can contact me at twitter https://twitter.com/misteroom

**Reply by officialrajdeepsingh** [reply]
Thanks @oscarotero

**Comment by officialrajdeepsingh** [comment]
Sorry I can use the following ways to add a relationship with the article and author. But it is not working.

```javascript

site.use(
  relations({
    foreignKeys: {
      article: "article_id",
      author: "author_name",
    }
  }));
```

```nunjucks

# posts/5-reasons-why-you-shouldn't-learn-photography-on-your-own.
# type="posts"
---
title: 5 Reasons Why You Shouldn't Learn Photography On Your Own.
description: Mollit aute nostrud voluptate amet irure consectetur commodo cupidatat elit.
  Non ut dolor nulla dolor duis. Anim eiusmod fugiat eiusmod ut nulla nulla
  labore.
date: 2022-11-08T08:18:10.494Z
draft: false
tags:
  - Photography
  - Learn Photography
category:
  - Photography
image: /images/camera.jpg


author_name: Jeanne Ballard

---

rest article here ...

```
```nunjucks
# author
# type="author"

---

author_name: Jeanne Ballard

description: Sit culpa elit excepteur ea Lorem ullamco eiusmod ex ut ea incididunt minim. Cillum eiusmod fugiat cupidatat.
date: 2022-11-08T09:01:53.077Z
draft: false
image: "/images/female1.jpg"
social: 
  - facebook.com
  - Google.com
job : "Writer"
---
```

**Reply by oscarotero** [reply]
I think it's cause `author_name` is the foreign key, so it must be used in the post, not in the author (that must be the id).

I'll try to explain better how it works:

- `idKey` The key used to identify a page. By default is `id`.
- `typeKey` The key used to define the type of a page. By default is `type`. 
- `foreign key`: The key used by other pages to relate themselves to a page.

The first two values are the same for all pages, and by default are `id` and `type`. This means all pages must have these two values. For example, your author could be `type=author` and `id=jeanne-ballard`.

The foreign key is different per type. For example, pages of type `author` may have the foreign key `author_id`. This is used by other pages to reference this page. For example, your post could have the variable `author_id=jeann-ballard`, so it's related with the page of type `author` and id `jeanne-ballard`.

Anyway, I'm working on improve this plugin to make it more easy to work with. Probably there will be changes in the incoming Lume 1.13.0.
